plugins/BEdita/Core/tests/TestCase/Model/Table/ObjectTypesTableTest.php

```diff
@@ -549,4 +551,47 @@ public function testParent($data)
         static::assertSame($parentId, $success->parent_id);
         static::assertTrue((bool)$success);
     }
+
+    /**
+     * Data provider for `testBeforeDelete`
+     *
+     * @return array
+     */
+    public function beforeDeleteProvider()
+    {
+        return [
+            'objects' => [
+                'objects',
+                new ForbiddenException('Abstract type with existing subtypes'),
+            ],
+            // there are no 'news` in objects fixture, safe to delete for now
+            'news' => [
+                'news',
+                true,
+            ],
+            'documents' => [
+                'documents',
+                new ForbiddenException('Objects of this type exist'),
+            ],
+        ];
+    }
+
+    /**
+     * Test `beforeDelete`
+     *
+     * @return void
+     * @dataProvider beforeDeleteProvider
+     * @covers ::beforeDelete()
+     * @covers ::beforeRules()
+     */
+    public function testBeforeDelete($typeName, $expected)
+    {
+        if ($expected instanceof \Exception) {
+            $this->expectException(get_class($expected));
+            static::expectExceptionMessage($expected->getMessage());
```

Comment:
This method _should_ be called dynamically (i.e.: `$this->expectExceptionMessage()`).

Reply:
Good to know...